core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/uri/ComponentURI.java

```diff
@@ -90,7 +90,7 @@ public String identifier() {
 
 	@JsonIgnore
 	public final boolean isUnknown() {
-		return UNKNOWN == this;
+		return UNKNOWN.equals(this);
```

Comment:
This was intentionally using `==`, because the original class was using an interner to cache the values.
I think a simple `Interners.weakInterner()` could be added to cache the instances and reduce the memory footprint of this value class.